modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/loader/java/ExtensionWithProvidedDependencyTestCase.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.loader.java;
+
+import static org.mule.runtime.module.extension.api.util.MuleExtensionUtils.loadExtension;
+import static org.mule.test.allure.AllureConstants.Sdk.MinMuleVersion.MIN_MULE_VERSION;
+import static org.mule.test.allure.AllureConstants.Sdk.SDK;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.api.meta.model.config.ConfigurationModel;
+import org.mule.runtime.api.meta.model.operation.OperationModel;
+import org.mule.test.provided.dependency.ProvidedDependencyExtension;
+
+import io.qameta.allure.Description;
+import io.qameta.allure.Feature;
+import io.qameta.allure.Issue;
+import io.qameta.allure.Story;
+import org.junit.Test;
+
+@Feature(SDK)
+@Story(MIN_MULE_VERSION)
+@Issue("W-14645134")
+public class ExtensionWithProvidedDependencyTestCase {
+
+  @Test
+  @Description("Tests that the model of an extension that has a custom type that relies on types from a dependency" +
+      "declared as provided, and has that custom type used as a parameter in an operation, is correctly loaded and" +
```

Comment:
I don't think this quite describes the issue. The provided class was not being used as a parameter but as an internal field. Also, in SDK language, a custom type usually refers to a type that's part of the extensions public API, while in this case it's an internal driver

Reply:
Done.